Source/WebKit/UIProcess/ios/forms/WKDateTimeInputControl.mm

```diff
@@ -192,25 +233,69 @@ - (NSString *)_sanitizeInputValueForFormatter:(NSString *)value
     return value;
 }
 
+- (RetainPtr<NSISO8601DateFormatter>)iso8601DateFormatterForCalendarView
+{
+    RetainPtr dateFormatter = adoptNS([[NSISO8601DateFormatter alloc] init]);
+    [dateFormatter setTimeZone:[NSTimeZone timeZoneForSecondsFromGMT:0]];
+    [dateFormatter setFormatOptions: NSISO8601DateFormatWithYear | NSISO8601DateFormatWithWeekOfYear | NSISO8601DateFormatWithDashSeparatorInDate];
+    return dateFormatter;
+}
+
 - (RetainPtr<NSDateFormatter>)dateFormatterForPicker
 {
     auto englishLocale = adoptNS([[NSLocale alloc] initWithLocaleIdentifier:@"en_US_POSIX"]);
     auto dateFormatter = adoptNS([[NSDateFormatter alloc] init]);
     [dateFormatter setTimeZone:[_datePicker timeZone]];
     [dateFormatter setDateFormat:_formatString];
+    // Necessary in order to append monday to the initial value for week inputs so that the NSDate reflects weeks beginning on Mondays in ISO-8601
+    if (_view.focusedElementInformation.elementType == WebKit::InputType::Week)
+        [dateFormatter setDateFormat:@"yyyy-'W'ww-EEEEE"];
     // Force English locale because that is what HTML5 value parsing expects.
     [dateFormatter setLocale:englishLocale.get()];
     return dateFormatter;
 }
 
 - (void)_dateChanged
 {
+#if HAVE(UI_CALENDAR_SELECTION_WEEK_OF_YEAR)
+    if (_view.focusedElementInformation.elementType == WebKit::InputType::Week) {
+        RetainPtr dateFormatter = [self iso8601DateFormatterForCalendarView];
+        [_view updateFocusedElementValue:[dateFormatter stringFromDate:[[NSCalendar calendarWithIdentifier:NSCalendarIdentifierISO8601] dateFromComponents:[_selectionWeekOfYear selectedWeekOfYear]]]];
+        return;
+    }
+#endif
+
     RetainPtr dateFormatter = [self dateFormatterForPicker];
     [_view updateFocusedElementValue:[dateFormatter stringFromDate:[_datePicker date]]];
 }
 
+#if HAVE(UI_CALENDAR_SELECTION_WEEK_OF_YEAR)
+
+- (void)setWeekPickerToInitialValue
+{
+    NSCalendarUnit unitFlags = NSCalendarUnitYearForWeekOfYear | NSCalendarUnitWeekOfYear | NSCalendarUnitWeekday;
+
+    if (![_initialValue length]) {
+        [_selectionWeekOfYear setSelectedWeekOfYear:[[NSCalendar calendarWithIdentifier:NSCalendarIdentifierISO8601] components:unitFlags fromDate:[NSDate date]]];
+        [self _dateChanged];
+        return;
+    }
+
+    // TODO: There may be a better way of ensuring the calculated NSDate from the initial value reflects ISO-8601 weeks starting on Monday.
```

Comment:
If we're going to merge this patch with this logic, this should be a `FIXME:` rather than a `TODO:`.

https://webkit.org/code-style-guidelines/#comments-fixme